libs/ui/src/pure/ExternalLink/index.tsx

```diff
@@ -55,61 +56,81 @@ export const LinkIcon = styled(LinkIconFeather as any)`
   stroke: currentColor;
 `
 
-export function handleClickExternalLink(event: React.MouseEvent<HTMLAnchorElement>) {
-  const { target, href } = event.currentTarget
-
-  const anonymizedHref = anonymizeLink(href)
-
-  // don't prevent default, don't redirect if it's a new tab
-  if (target === '_blank' || event.ctrlKey || event.metaKey) {
-    outboundLink({ label: anonymizedHref }, () => {
-      console.debug('Fired outbound link event', anonymizedHref)
-    })
-  } else {
-    event.preventDefault()
-    // send a ReactGA event and then trigger a location change
-    outboundLink({ label: anonymizedHref }, () => {
-      window.location.href = anonymizedHref
-    })
-  }
-}
+export type ExternalLinkProps = Omit<HTMLProps<HTMLAnchorElement>, 'as' | 'ref' | 'onClick'> & {
+  href: string
+  onClickOptional?: React.MouseEventHandler<HTMLAnchorElement>
+} & PropsWithChildren
 
 /**
  * Outbound link that handles firing google analytics events
  */
 export function ExternalLink({
+  children,
   target = '_blank',
   href,
   rel = 'noopener noreferrer',
   onClickOptional,
   ...rest
-}: Omit<HTMLProps<HTMLAnchorElement>, 'as' | 'ref' | 'onClick'> & {
-  href: string
-  onClickOptional?: React.MouseEventHandler<HTMLAnchorElement>
-}) {
+}: ExternalLinkProps) {
+  const cowAnalytics = useCowAnalytics()
   return (
     <StyledLink
       target={target}
       rel={rel}
       href={href}
       onClick={(event) => {
         if (onClickOptional) onClickOptional(event)
-        handleClickExternalLink(event)
-        externalLinkAnalytics(href)
+        handleClickExternalLink(cowAnalytics, event)
+        externalLinkAnalytics(cowAnalytics, href)
       }}
       {...rest}
-    />
+    >
+      {children}
+    </StyledLink>
   )
 }
 
+function handleClickExternalLink(cowAnalytics: CowAnalytics, event: React.MouseEvent<HTMLAnchorElement>): void {
+  const { target, href } = event.currentTarget
+
+  const anonymizedHref = anonymizeLink(href)
+
+  const isNewTab = target === '_blank' || event.ctrlKey || event.metaKey
+
+  // don't prevent default, don't redirect if it's a new tab
+  if (!isNewTab) {
+    event.preventDefault()
+  }
+
+  cowAnalytics.outboundLink({
+    label: anonymizedHref,
+    hitCallback: () => {
+      if (isNewTab) {
+        console.debug('Fired outbound link event', anonymizedHref)
+      } else {
+        // send a ReactGA event and then trigger a location change
+        window.location.href = anonymizedHref
+      }
+    },
+  })
+}
+
 export function ExternalLinkIcon({
   target = '_blank',
   href,
   rel = 'noopener noreferrer',
   ...rest
 }: Omit<HTMLProps<HTMLAnchorElement>, 'as' | 'ref' | 'onClick'> & { href: string }) {
+  const cowAnalytics = useCowAnalytics()
```

Comment:
I know, I know, but this is a pure component.
Should not have side effects/hooks.

Reply:
599906b707d5d0b8e9ca845b8c3069ccd3c8baa5